eth/catalyst/simulated_beacon.go

```diff
@@ -156,6 +164,16 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal, timestamp u
 		c.setCurrentState(header.Hash(), *finalizedHash)
 	}
 
+	// Because transaction insertion, block insertion and block production will
+	// happen without any timing delay between them in simulator mode and the
+	// transaction pool will be running its internal reset operation on a
+	// background thread, flaky executions can happen. To avoid the racey
+	// behavior, the pool will be explicitly blocked on its reset before
+	// continuing to the block production below.
+	if err := c.eth.APIBackend.TxPool().Sync(); err != nil {
```

Comment:
I don't think we need a manual call to `Sync()` here.   When we are running in non-zero period dev mode, any includeable transactions that weren't picked up will be caught when sealing a subsequent block.

Reply:
Oh right.  This is for tests.